### DISCUSSION TITLE
Time Node (Beta Version)

### DISCUSSION THREAD

**Discussion by zachowj** [zachowj/node-red-contrib-home-assistant-websocket#380] [General]
place holder

**Comment by hvddrift** [comment]
Would you please have an example of the 'Time Node'?

I would like to test it, but cant find the right combinations.

**Reply by zachowj** [reply]
Check out the cookbook <https://zachowj.github.io/node-red-contrib-home-assistant-websocket/cookbook/using-date-and-time-entities-to-trigger-flows.html>.

I converted several flows to use the time node.

**Comment by arretx** [comment]
Lol...FYI, the statement "Beta version: Config should be stable and hopefully not to many bugs.
Found an issue? Post it in issues . Have questions or comments? Post them here ." contains a bug itself.  not "too" many bugs.

**Comment by rikki78** [comment]
Nice feature this time node, saves some additional nodes after an input_date_time. A nice feature would be if the days when this event is fired can be selected somewhere.

**Comment by Michelobb** [comment]
Hello and thanks for this great node.

I got a strange issue.

I'm using the node to set my home status (day or night) in an INPUT_SELECT HA variable.

The node is supposed to fire at a scheduled daily time, stored in an INPUT_NUMBER HA variable.

Things work remarkably well in the evening, when the house'status changes from "day" to "night".

But in the morning, it hasn't yet worked a single time.

However, if I simulate being in the morning, setting the HA variable to the next minute, in order to be able to check it, the change occurs perfectly well.

So, do I have to be awake to make it work ?  (haha)

Seriously, should I trigger some event before this one, just to "wake up" the system ?  This doesn't sound quite rational, does it ? 

Any help would be really appreciated here.

Thanks,

Michel

`[{"id":"083494addd03ef2a","type":"api-call-service","z":"20594c77ddba82b2","name":"Mode jour !","server":"e5371104.417d5","version":3,"debugenabled":false,"service_domain":"input_select","service":"select_option","entityId":"input_select.mode_maison","data":"{\"option\":\"Jour\"}","dataType":"jsonata","mergecontext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":450,"y":600,"wires":[[]]},{"id":"75ee502e6f5e913c","type":"ha-time","z":"20594c77ddba82b2","name":"Heure du mode Jour ?","server":"e5371104.417d5","version":0,"exposeToHomeAssistant":false,"haConfig":[{"property":"name","value":""},{"property":"icon","value":""}],"entityId":"input_datetime.mode_jour_time","property":"state","offset":0,"offsetType":"num","offsetUnits":"minutes","randomOffset":false,"repeatDaily":false,"payload":"$entity().state","payloadType":"jsonata","debugenabled":false,"x":190,"y":600,"wires":[["083494addd03ef2a"]]},{"id":"e5371104.417d5","type":"server","name":"Home Assistant","version":1,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":true}]`

**Reply by zachowj** [reply]
My only guess at the moment might be that the connection to HA was interrupted before the call service in the morning. Do you have any errors in the NR logs about call-service called without connection to the server? If this is the case you can turn on queueing in the call-service node and it will update the input select when it connects to HA.

**Reply by Michelobb** [reply]
Hello and thanks for your reply.

I made the change this night, and the system worked very well this morning.  So many thanks for this.

I'm still a bit surprised because as it didn't work about 5 ou 6 days, that would mean NR was disconnected every day at the same time from HA...  I doubt it, but anyway, your hopefully reminded me 1. the queuing feature, which is great (I was about to use MQTT to ensure communication) and 2. ALWAYS look to the logs when things go wrong... which I didn't.  So from now, next time (if) the issue comes back, I'll check this out.

Thanks again,

Michel

**Reply by Michelobb** [reply]
Got some news.

The automation doesn't work in the morning since 2 days (still no problem at night).

So yesterday I switched the NR log level to Debug, and this morning, I can see that the Time node didn't trigger at all :  I got no error message in the log (about connexion or anything), and in the NR GUI, the node clearly shows its last trigger time :  about 2 days ago (wich is in fact my last "manual" try).

On its side, the Call Service node shows correct triggering yesterday evening (because the "evening Time node" works well).

Sounds like a whole mystery to me.

Would you need my logs or something ?

Thanks,

Michel

**Reply by zachowj** [reply]
Do you have `Repeat Daily` checked?

**Reply by Michelobb** [reply]
No I don't, but on the "evening" node neither.

Now I've checked it, and I'll see what's happenning tomorrow, but then, I don't understand why the evening node is working without that...

**Reply by Michelobb** [reply]
Now that I believe having understood the node's logic, I've also checked my HA input_select variables, and they both send HH:MM:SS without DAY.  I thought this could 've been responsible of the divergent behavior, but it's not.

**Reply by zachowj** [reply]
Are you updating the evening input_select daily by some means?

**Reply by Michelobb** [reply]
I don't...  I only changed manually a couple of time, to test it.  But it seems to fire even if I don't touch it the whole day.

Something I'm not sure, but I think having seen the mention "In the past" only on the morning node status.  But that's something I couldn' t certify.

**Reply by Michelobb** [reply]
I think I need to check very carefully a whole new day without updating the evening node time after its first shot, just to be sure we can discard that possibility.
